### PR TITLE
feat: Complete Changes module

### PR DESCRIPTION
**Pull request by L-390**
issue #1322

### CODE REVIEW

**Review by scopsy — COMMENTED** [review]
there was a few missing methods types needs to be changed based on the swagger docs :)

Could you please take a look at https://docs.novu.co/api/apply-changes/ to validate the contents?

**Comment by florian-lefebvre** [comment]
#1357 has bien merged into main, please pull the changes. You'll have to extend thé WithHttp classique but I think that's all

**Comment by L-390** [comment]
I hope everything went alright when pulling it because it was a huge struggle to pull the new changes :)

File: packages/node/src/lib/changes/changes.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    return await this.http.post(`/v1/changes/${changeId}/apply`, {});
+  }
+
+  /**
+   * @param {string[]} changeIds - An Array of changes that will be applied
+   * @returns {promise<object>} - Applies all changes specified in the array
+   */
+  async applyMany(changeIds: string[]) {
+    return await this.http.post(`/v1/changes/bulk/apply`, { changeIds });
+  }
```
Comment (scopsy) [inline packages/node/src/lib/changes/changes.ts:42]: @davidsoderberg do you think we have those types available at @novu/shared?

File: packages/node/src/lib/changes/changes.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  applyAll() {
+    throw new Error('Method not implemented.');
+  }
+
+  /**
+   * @returns {promise<object>} - Returns an object containing all changes
+   */
+  async get() {
+    return await this.http.get(`/v1/changes`);
```
[the remark this answers is not in the document]
Comment (florian-lefebvre) [reply packages/node/src/lib/changes/changes.ts:18]: I haven't changed anything about this but you're right, v1 is not required (see https://github.com/novuhq/novu/blob/main/packages/node/src/lib/novu.ts#L34)

File: packages/node/src/lib/changes/changes.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
+import { AxiosInstance } from 'axios';
+import { IChanges } from './changes.interface';
+
+export class Changes implements IChanges {
```
Comment (florian-lefebvre) [inline packages/node/src/lib/changes/changes.ts:4]: It must be `export class Changes extends WithHttp implements iChanges {`. This means you don't have to set the private http nor writing the constructor.

File: packages/node/src/lib/changes/changes.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
+import { AxiosInstance } from 'axios';
```
Comment (florian-lefebvre) [inline packages/node/src/lib/changes/changes.ts:1]: Replace with `import axios from 'axios',`. It won't be used but it's necessay for types coming with WithHttp

**Review by florian-lefebvre — CHANGES_REQUESTED** [review]

**Comment by L-390** [comment]
Alright I added the "withHTTP" and removed the /v1, at least if I understood everything correctly :)

File: packages/node/src/lib/changes/changes.ts [hunk 5]
```diff
@@ -0,0 +1,39 @@
+import AxiosInstance from 'axios';
```
Comment (florian-lefebvre) [inline packages/node/src/lib/changes/changes.ts:1]: replace AxiosInstance with axios ;)

**Review by florian-lefebvre — CHANGES_REQUESTED** [review]

**Review by scopsy — APPROVED** [review]

**Comment by nilesh-maurya** [comment]
@L-390 Tests related to `Changes module` are failing now. can you update those ?

File: packages/node/src/lib/changes/changes.spec.ts [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    novu = new Novu(mockConfig.apiKey);
+  });
+
+  test('should get changes correctly', async () => {
+    mockedAxios.post.mockResolvedValue({});
+
+    await novu.changes.get();
+
+    expect(mockedAxios.get).toHaveBeenCalled();
+    expect(mockedAxios.get).toHaveBeenCalledWith('/v1/changes');
```
Comment (nilesh-maurya) [inline packages/node/src/lib/changes/changes.spec.ts:25]: here it will be `/changes` instead of `/v1/changes`,  same goes other tests also.

**Pull request merged**